app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/mapper/DetailsMapper.kt

```diff
@@ -1,83 +1,87 @@
+            agendaItem = AgendaItemDetails.Task(
+                isDone = isDone
+            ),
         )
-        is AgendaItem.EventItem -> AgendaDetailsState(
-            id = event.id ?: UUID.randomUUID().toString(),
-            title = event.title,
-            description = event.description ?: "",
-            fromTime = event.from.toLocalTime(),
-            fromDate = event.from.toLocalDate(),
+        is AgendaItem.Event -> AgendaDetailsState(
+            id = id,
+            title = title,
+            description = description ?: "",
+            fromTime = from.toLocalTime(),
+            fromDate = from.toLocalDate(),
+            reminderType = getReminderType() ?: ReminderType.ThirtyMinutes,
             agendaItem = AgendaItemDetails.Event(
-                toTime = event.to.toLocalTime(),
-                toDate = event.to.toLocalDate(),
+                toTime = to.toLocalTime(),
+                toDate = to.toLocalDate(),
             ),
         )
-        is AgendaItem.ReminderItem -> AgendaDetailsState(
-            id = reminder.id ?: UUID.randomUUID().toString(),
-            title = reminder.title,
-            description = reminder.description ?: "",
-            fromDate = reminder.time.toLocalDate(),
-            fromTime = reminder.time.toLocalTime(),
+        is AgendaItem.Reminder -> AgendaDetailsState(
+            id = id,
+            title = title,
+            description = description ?: "",
+            fromDate = from.toLocalDate(),
+            fromTime = from.toLocalTime(),
+            reminderType = getReminderType() ?: ReminderType.ThirtyMinutes,
             agendaItem = AgendaItemDetails.Reminder
         )
     }
 }
 
-fun LocalDate.toFormatted(): String {
-    val dateInLocalZone = this
-        .atStartOfDay(ZoneId.systemDefault())
-    return DateTimeFormatter
-        .ofPattern("MMM dd yyyy")
-        .format(dateInLocalZone)
-}
-
-fun LocalTime.toFormatted(): String {
-    return DateTimeFormatter
-        .ofPattern("HH:mm")
-        .format(this)
-}
-
-val AgendaDetailsState.eventDateTime: LocalDateTime
-    get() =  fromDate.atTime(fromTime)
-
-fun ReminderType.toReminderText(): UiText {
-    return when(this) {
-        ReminderType.TenMinutes -> UiText.StringResource(id = R.string.ten_minutes_before)
-        ReminderType.ThirtyMinutes -> UiText.StringResource(id = R.string.thirty_minutes_before)
-        ReminderType.OneHour -> UiText.StringResource(id = R.string.one_hour_before)
-        ReminderType.SixHours -> UiText.StringResource(id = R.string.six_hours_before)
-        ReminderType.OneDay -> UiText.StringResource(id = R.string.one_day_before)
+fun AgendaDetailsState.toAgendaItem(): AgendaItem {
+    val remindAt = reminderType.toReminderDateFromDateTime(fromDate.atTimeToUtc(fromTime))
+    val from = fromDate.atTimeToUtc(fromTime)
+    return when (agendaItem) {
+        is AgendaItemDetails.Task -> AgendaItem.Task(
+            id = id ?: UUID.randomUUID().toString(),
```

Comment:
When you can generate the ID locally, there is no reason that it should be nullable. You are just inviting bugs into your code 🐛 `null` has its place, but think twice if it is something you need before using it.

Reply:
I'll keep that in mind, thank you!